Allow dotcms dev container to start without a license

### Parent Issue

_No response_

### Problem Statement

running `just dev-run` starts dotCMS but it will fail with an error if there is no available license.

```log
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  12.979 s
[INFO] Finished at: 2024-02-28T13:22:01-05:00
[INFO] ------------------------------------------------------------------------
[WARNING]
[WARNING] Plugin validation issues were detected in 2 plugin(s)
[WARNING]
[WARNING]  * org.apache.maven.plugins:maven-enforcer-plugin:3.0.0
[WARNING]  * io.fabric8:docker-maven-plugin:0.43.4
[WARNING]
[WARNING] For more or less details, use 'maven.plugin.validation' property with one of the values (case insensitive): [BRIEF, DEFAULT, VERBOSE]
[WARNING]
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-antrun-plugin:3.1.0:run (cleanup-it-test-data) on project dotcms-core: An Ant BuildException has occured: The following error occurred while executing this line:
[ERROR] /Users/will/git/dotcms/core/dotCMS/ant-tasks.xml:17: The following error occurred while executing this line:
[ERROR] /Users/will/git/dotcms/core/dotCMS/ant-tasks.xml:51: /Users/will/.dotcms/license does not exist.
[ERROR] around Ant part ...<ant antfile="/Users/will/git/dotcms/core/dotCMS/ant-tasks.xml">... @ 4:69 in /Users/will/git/dotcms/core/dotCMS/target/antrun/build-main.xml
[ERROR] -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
error: Recipe `dev-tomcat-run` failed on line 87 with exit code 1
```


### Steps to Reproduce

checkout dotcms from github
from the project folder, try to run just dev-run

### Acceptance Criteria

dotcms should start normally, even without a license.

### dotCMS Version

24.02

### Proposed Objective

Technical User Experience

### Proposed Priority

Priority 2 - Important

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.

_No response_

### Assumptions & Initiation Needs

_No response_

### Quality Assurance Notes & Workarounds

_No response_

### Sub-Tasks & Estimates

_No response_

--- a/dotCMS/ant-tasks.xml
+++ b/dotCMS/ant-tasks.xml
@@ -16,7 +16,6 @@
 
         <antcall target="create-test-folder" inheritAll="true" inheritRefs="true">
         </antcall>
-
     </target>
 
     <target name="create-test-folder">
@@ -46,13 +45,16 @@
                 <echo file="${context.path}/${test.dynamic.folder}/license/license.dat" append="false" message="${env.DOT_DOTCMS_LICENSE}"/>
             </then>
             <else>
-                <echo>Copying license from ${user.home}/.dotcms/license</echo>
-                <copy todir="${context.path}/${test.assets.folder}">
-                    <fileset dir="${user.home}/.dotcms/license" includes="license.zip" />
-                </copy>
+                <ac:if>
+                    <available file="${user.home}/.dotcms/license/license.zip"/>
+                    <then>
+                        <copy todir="${context.path}/${test.assets.folder}">
+                            <fileset dir="${user.home}/.dotcms/license" includes="license.zip" />
+                        </copy>
+                    </then>
+                </ac:if>
             </else>
         </ac:if>
-
     </target>
 
     <target name="cleanup-test-folders">

